client/components/domains/domain-registration-suggestion/index.jsx

```diff
@@ -173,6 +174,9 @@ class DomainRegistrationSuggestion extends React.Component {
 				buttonStyles = { ...buttonStyles, disabled: true };
 			}
 		}
+		if ( this.props.isCartPendingUpdate ) {
+			buttonStyles = { ...buttonStyles, busy: true };
```

Comment:
I think this check should go higher, to line 170 and simply be `} else if ( pendingCheckSuggestion || this.props.isCartPendingUpdate ) {`. Otherwise, this causes a weird UX where suddenly the suggestions `Select` buttons go from disabled to suddenly busy _and_ clickable. So, you can actually select another domain while you're waiting to be redirected to the cart. Which is not an error in itself, but I can imagine could cause some confusion (and possibly unwanted domains being bought).

Reply:
That's a good suggestion; I didn't realize that if `busy` is set, the button remains clickable!

In that case, we'll need to decide what to do about the existing `if ( pendingCheckSuggestion )` clause. With your suggestion, that clause will never activate, which makes me think that you're saying we should _replace_ line 170 with the condition. But if we do that, then the button will not display as `busy` when adding the product to the cart, it will be `disabled` instead because `pendingCheckSuggestion` will not be set. I wonder if you mean removing the nested `if` entirely like:

```js
		} else if ( pendingCheckSuggestion || this.props.isCartPendingUpdate ) {
			buttonStyles = { ...buttonStyles, busy: true, disabled: true };
		}
```

Reply:
Or maybe:

```js
		} else if ( pendingCheckSuggestion || this.props.isCartPendingUpdate ) {
			if ( pendingCheckSuggestion.domain_name === suggestion.domain_name ) {
				buttonStyles = { ...buttonStyles, busy: true };
			} else {
				buttonStyles = { ...buttonStyles, busy: true, disabled: true };
			}
		}
```